### PR TITLE
Build container rpms from docker repositories

### LINKED ISSUE #162: Pre-loading container images into CRI-O
MicroShift needs a built-in mechanism to pre-load container images into nodes' CRI-O instances so that it can bootstrap its component services a) without network connectivity and b) without local image registry.

The mechanism should monitor `/var/lib/microshift/images` for oci-archives or multi-image docker-archives as produced by `podman save` and pre-load them into CRI-O on cluster initialization or whenever that directory changes.

Requirements:
* MUST NOT change image digests, so the images can be referenced from K8s manifests by digest, too.
* MUST be compatible with multi-arch images and having multiple tarballs/versions of images (e.g. for updates)
* SHOULD unzip tarballs if necessary.
* SHOULD support deleting pre-loaded images from `/var/lib/microshift/images` and subsequently pruning from CRI-O.

### PR DESCRIPTION
**Pull request by mangelajo**
This will ease integration on build systems like copr, those
build systems build as a non privileged user where podman pull --root
won't work.

Also, this packs a tarball file in the rpm which is extracted during
the post phase to avoid additional limitations of the build systems
removing the tarball uid/gids of the files. Also cpio storage
in the rpm will save all as root/root unless explicitly told otherwise
via %files directives.

**Which issue(s) this PR addresses**:

Closes #162 

Co-Authored-By:	Brent Salisbury <bsalisbu@redhat.com>

### CODE REVIEW

File: packaging/rpm/make-microshift-images-rpm.sh [hunk 1]
```diff
@@ -23,26 +26,33 @@ ARCHITECTURES=${ARCHITECTURES:-"x86_64 arm64 arm ppc64le riscv64"}
 build() {
   cat >"${RPMBUILD_DIR}"SPECS/microshift-images.spec <<EOF
 %global baseVersion ${BASE_VERSION}
-
+%global github_tag ${GITHUB_TAG}
 EOF
```
Comment (mangelajo) [inline packaging/rpm/make-microshift-images-rpm.sh:30]: WIP: this all needs to be changed into generating a .yaml file, and calling paack.py

**Comment by mangelajo** [comment]
rebased on main

File: packaging/rpm/paack.py [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+                                    '%s-%s-%s.src.rpm' % (package['name'], package['version'], package['release']))
+
+            # if we are in no-cleanup mode the existing srpm is reused instead
+            if not (os.path.exists(out_srpm) and self._no_cleanup):
+                spec = SpecFile(package['name'], os.path.join(self._rpmbuild_dir, 'SPECS',
+                    '%s-%s-%s.spec' % (package['name'], package['version'], package['release'])))
+                for arch_info in package['arch']:
+                    arch = arch_info['name']
+                    tar_path = self._create_tarball(package, arch_info)
+                    print("scanning %s to generate files section for arch %s" %(tar_path, arch))
```
Comment (oglok) [inline packaging/rpm/paack.py:240]:
```suggestion
                    print("scanning %s to generate files section for arch %s" % (tar_path, arch))
```
nit: space for style consistency

**Review by oglok — COMMENTED** [review]
/lgtm

Just a nit

**Pull request merged**